src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java

```diff
@@ -0,0 +1,244 @@
+        }
+    }
+
+    //Task 4
+    public String getTrimSpaceToLowerCase(String str) {
+        return str == null ? null : str.trim().toLowerCase();
+    }
+
+    //Task 5
+    // excluding second symbol
+    public String getSubstring(String str, char firstSymbol, char secondSymbol) {
+        int indexFirstSymbol = getIndexOfSymbol(str, firstSymbol);
+        int indexSecondSymbol = getIndexOfSymbol(str, secondSymbol);
+        return indexFirstSymbol < 0 || indexSecondSymbol < 0 ? "" : str.substring(indexFirstSymbol, indexSecondSymbol);
+    }
+
+    //Task 6
+    public String replace(String str, String oldValue, String newValue) {
+        if (str == null) {
+            return null;
+        } else if (oldValue == null) {
+            return str;
+        } else if (newValue == null) {
+            newValue = "";
+        }
+        return str.replace(oldValue, newValue);
+    }
+
+    //Task 7
+    public boolean isStartAndEndWithWord(String text, String word) {
+        return text != null && word != null && text.startsWith(word) && text.endsWith(word);
+    }
+
+    //Task 8
+    public int countVowels(String str) {
+        int amountVowels = 0;
+        if (str == null) {
+            return amountVowels;
+        }
+        char[] vowels = {'a', 'e', 'i', 'o', 'u', 'y'};
+        char[] strArray = str.toLowerCase().toCharArray();
+        for (char c : strArray) {
+            for (char vowel : vowels) {
+                if (c == vowel) {
+                    amountVowels++;
+                }
+            }
+        }
+        return amountVowels;
+    }
+
+    //Task 9
+    public int countPunctuationMarks(String str) {
+        int amountPunctuationMarks = 0;
+        if (str == null) {
+            return amountPunctuationMarks;
+        }
+        char[] punctuationMarks = {',', '.', '!', '?'};
+        char[] strArray = str.toCharArray();
+        for (char c : strArray) {
+            for (char punctuationMark : punctuationMarks) {
+                if (c == punctuationMark) {
+                    amountPunctuationMarks++;
+                }
+            }
+        }
+        return amountPunctuationMarks;
+    }
+
+    //Task 10
+    public boolean isPalindrome(String str) {
+        if (str == null) {
+            return false;
+        }
+        String modified = str.toLowerCase().replaceAll("\\W", "");
+        StringBuilder stringBuilder = new StringBuilder(modified);
+        String inverted = stringBuilder.reverse().toString();
+        return isEqual(modified, inverted);
+    }
+
+    //Task 11
+    public String[] splitToArray(String str, int numberSymbols) {
+        if (str == null) {
+            return null;
+        }
+        int length = 0;
+        if (str.length() % numberSymbols == 0) {
+            length += str.length() / numberSymbols;
+        } else {
+            length += str.length() / numberSymbols + 1;
+        }
+        String[] array = new String[length];
```

Comment:
useless code

Reply:
Could you please explain why is useless, cause without this condition if I have even amount of symbols in my String I get StringIndexOutOfBoundsException Error.